test/unit/core/viz/expressions/ramp.test.js

```diff
@@ -17,19 +20,374 @@ describe('src/core/viz/expressions/ramp', () => {
+                });
+
+                it('should get the second value', () => {
+                    const r = ramp(1, [firstColor, secondColor]);
+
+                    r._compile();
+                    actual = r.eval();
+                    expected = secondColor._nameToRGBA();
+
+                    expect(actual).toEqual(expected);
+                });
+            });
+        });
+    });
+
+    describe('.eval with buckets', () => {
+        describe('when palettes are color arrays', () => {
+            const METADATA = {
+                columns: [
+                    {
+                        name: 'grade',
+                        type: 'category',
+                        categoryNames: [
+                            'A',
+                            'B',
+                            'C',
+                            'D',
+                            'E'
+                        ],
+                    },
+                ],
+                categoryIDs: {
+                    'A': 'A',
+                    'B': 'B',
+                    'C': 'C',
+                    'D': 'D',
+                    'E': 'E'
+                }
+            };
+            
+            const red = s.namedColor('red');
+            const blue = s.namedColor('blue');
+            const yellow = s.namedColor('yellow');
+            const purple = s.namedColor('purple');
+            const green = s.namedColor('green');
+            const orange = s.namedColor('orange');
+
+            let actual;
+            let expected;
+
+            describe('and there are less categories than colors', () => {
+                describe('with "others"', () => {
+                    it('should not show interpolation', () => {
+                        const r = ramp(buckets('B', ['A', 'B', 'C']), [red, blue, yellow, purple]);
+
+                        r._compile(METADATA);
+                        actual = r.eval();
+                        expected = blue._nameToRGBA();
+        
+                        expect(actual).toEqual(expected);
+                    });
+
+                    it('should use last color for "others"', () => {
+                        const r = ramp(buckets('D', ['A', 'B', 'C']), [red, blue, yellow, purple]);
+
+                        r._compile(METADATA);
+                        actual = r.eval();
+                        expected = purple._nameToRGBA();
+        
+                        expect(actual).toEqual(expected);
+                    });
+                });
+
+                describe('without "others"', () => {
+                    it('should show interpolation', () => {
+                        const r = ramp(buckets('C', ['A', 'B', 'C', 'D', 'E']), [red, blue, yellow, purple, green, orange]);
+
+                        r._compile(METADATA);
+                        actual = r.eval();
+                        expected = yellow._nameToRGBA();
+        
+                        expect(actual).not.toEqual(expected);
```

Comment:
This assertion is not strong enough, can we expect the interpolated color instead?

Reply:
I will think a better test for this case, but the answer is no, we can not expect the interpolated color. First, because we do not care about the precise interpolated value, and second, because it would be a weak test case, hard to understand and that we can break it easily if we change the way we interpolate.

Reply:
we can always use [toBeCloseTo](https://jasmine.github.io/api/2.6/matchers.html#toBeCloseTo) with a threshold

Reply:
I like that!